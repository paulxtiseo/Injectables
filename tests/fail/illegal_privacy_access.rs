use injectables::{injectable, inject_fields};

mod other_module {
    use injectables::{injectable, inject_fields};

    #[injectable]
    pub struct Private {
        id: u64,           // private field
        pub name: String,  // public field
    }
}

#[inject_fields(other_module::Private)]
pub struct IllegalAccess {
    pub value: u32,
}

fn main() {
    let test = IllegalAccess {
        value: 42,
        id: 1,      // this should fail to compile - cannot access private field
        name: "Test".to_string(),
    };

    // use the variable to prevent unused variable warnings
    println!("name: {}", test.name);
}