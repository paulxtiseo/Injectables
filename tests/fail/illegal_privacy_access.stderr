error: Cannot access field 'id' with visibility Private from module 'other_module' in module ''
  --> tests/fail/illegal_privacy_access.rs:13:1
   |
13 | #[inject_fields(other_module::Private)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0422]: cannot find struct, variant or union type `IllegalAccess` in this scope
  --> tests/fail/illegal_privacy_access.rs:19:16
   |
19 |     let test = IllegalAccess {
   |                ^^^^^^^^^^^^^ not found in this scope

warning: unused import: `injectable`
 --> tests/fail/illegal_privacy_access.rs:1:19
  |
1 | use injectables::{injectable, inject_fields};
  |                   ^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `inject_fields`
 --> tests/fail/illegal_privacy_access.rs:4:35
  |
4 |     use injectables::{injectable, inject_fields};
  |                                   ^^^^^^^^^^^^^
